ext/drx/scatter_gather_aarch64.c

```diff
@@ -481,47 +493,60 @@ expand_scatter_gather(void *drcontext, instrlist_t *bb, instr_t *sg_instr,
                            DR_PRED_SVE_NONE),
                 orig_app_pc));
 
-        const reg_id_t reg_to_extract =
-            reg_is_z(sg_info->base_reg) ? sg_info->base_reg : sg_info->index_reg;
-
-        /* lastb    scratch_gpr0, scratch_pred, reg_to_extract.element_size */
-        EMIT(lastb_sve_scalar, opnd_create_reg(scratch_gpr0),
-             opnd_create_reg(scratch_pred),
-             opnd_create_reg_element_vector(reg_to_extract, sg_info->element_size));
-
-        const reg_id_t scalar_index_or_base = scratch_gpr0;
+        for (uint reg_index = 0; reg_index < sg_info->reg_count; reg_index++) {
+            if (reg_index == 0) {
+                const reg_id_t reg_to_extract =
+                    reg_is_z(sg_info->base_reg) ? sg_info->base_reg : sg_info->index_reg;
+
+                /* lastb    scalar_index_or_base, scratch_pred,
+                 * reg_to_extract.element_size */
+                EMIT(lastb_sve_scalar, opnd_create_reg(scalar_index_or_base),
+                     opnd_create_reg(scratch_pred),
+                     opnd_create_reg_element_vector(reg_to_extract,
+                                                    sg_info->element_size));
+            } else {
+                // Increment index
```

Comment:
Ah this is where the magic happens where the ith byte is loaded to the ith vector. The comment doesn't do it justice. Please elaborate.

Reply:
Done.